manager/ui/war/plugins/api-manager/ts/api/api-def.ts

```diff
@@ -32,39 +32,68 @@ module Apiman {
 
             $scope.isEntityDisabled = EntityStatusSvc.isEntityDisabled;
 
-            $scope.saveApi = function() {
+            let isValidDefinition = function (data, definitionType) {
+                switch (definitionType) {
+                    case 'SwaggerJSON':
+                        try {
+                            $.parseJSON(data);
+                            return true;
+                        } catch (error) {
+                            return false;
+                        }
+                    case 'WSDL':
+                        try {
+                            $.parseXML(data);
+                            return true;
+                        } catch (error) {
+                            return false;
+                        }
+                        default:
+                            return true;
+                }
+            }
+
+            $scope.saveApi = function () {
                 console.log('$scope.selectedDefinitionType.value: ' + $scope.selectedDefinitionType.value);
                 $scope.saveButton.state = 'in-progress';
 
+                if (isValidDefinition($scope.updatedApiDefinition, $scope.selectedDefinitionType.value) == false) {

```

Comment:
I would prefer if (!isValidDefinition($scope.updatedApiDefinition, $scope.selectedDefinitionType.value)) instead of comparing the boolean value with false